### PR TITLE
Remove user creation and modernize

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request removes user creation from the configuration and modernizes the rest of the configuration.

## 💭 Motivation and context ##

User creation should now be managed by [cisagov/cyhy-users-non-admin]. I modernized the repo a bit while I was here just to make it more consistent with how we do similar configurations in COOL projects.

## 🧪 Testing ##

Automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Migrate users to [cisagov/cyhy-users-non-admin] and remove them from the state.
- [x] Destroy this configuration as it is deployed.
- [x] Do a fresh apply to lay down the policies using the new configuration.

[cisagov/cyhy-users-non-admin]: https://github.com/cisagov/cyhy-users-non-admin

### CODE REVIEW

File: locals.tf [hunk 1]
```diff
@@ -5,8 +5,18 @@
...
+# Retrieve information about the findings data bucket we will be providing access
+# for with this workspace.
+data "aws_s3_bucket" "findings_data" {
+  bucket = local.findings_data_s3_bucket
+}
+
 locals {
   # This is a goofy but necessary way to determine if
   # terraform.workspace contains the substring "prod"
   production_workspace = replace(terraform.workspace, "prod", "") != terraform.workspace
```
Comment (dav3r) [inline locals.tf]: As long as we are cleaning up, let's use a regex here:
```suggestion
  # Determine if this is a Production workspace by checking
  # if terraform.workspace begins with "prod-"
  production_workspace =  length(regexall("^prod-", terraform.workspace)) == 1
```

**Review by dav3r — APPROVED** [review]
This looks great to me. 👍 

I also noted one line to be modernized that caught my eye.

File: locals.tf (hunk 1 above)
Comment (mcdonnnj) [reply locals.tf]: Ah, much nicer. Glad you noticed!

File: README.md [hunk 2]
```diff
@@ -100,8 +99,11 @@ No modules.
...
+| findingsbucketfullaccess\_policy\_description | The description to associate with the IAM policy that allows full access to the findings data S3 bucket. | `string` | `"Allows full access to the S3 bucket where findings data is stored."` | no |
+| findingsbucketfullaccess\_policy\_name | The base name to associate with the IAM policy that allows full access to the findings data S3 bucket.  Note that in production workspaces, '-production' is automatically appended this bucket name.  In non-production workspaces, '-<workspace\_name>' is automatically appended to this bucket name. | `string` | `"FindingsBucketFullAccess"` | no |
 | tags | Tags to apply to all AWS resources created. | `map(string)` | `{}` | no |
-| usernames | The usernames associated with the accounts to be created.  The format first.last is recommended. | `list(string)` | n/a | yes |
+| users | A list of the usernames for the users that should be given access to the S3 bucket that manages Assessment findings data. | `list(string)` | n/a | yes |
```
Comment (jsf9k) [inline README.md:110]: Is it worth mentioning that these users _must_ have already been created via [cisagov/cool-users-non-admin](https://github.com/cisagov/cool-users-non-admin)?

**Review by jsf9k — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (mcdonnnj) [reply README.md:110]: This is a good point, but I would rather bake it into the pre-requisites in the README. Thoughts?

Comment (dav3r) [reply README.md:110]: I'm fine as long as it's mentioned somewhere.

Comment (jsf9k) [reply README.md:110]: I don't think it hurts to mention it in both places, but I will defer to your judgement as long as it is mentioned somewhere.

Comment (mcdonnnj) [reply README.md:110]: The COOL -iam configurations only mention it as a pre-requisite in the README (which is what I'm using as a "standard").

Comment (mcdonnnj) [reply README.md:110]: Please see [733c715](https://github.com/cisagov/findings-data-s3-users/pull/21/commits/733c715116291cc9aea398bada4b3fff2845537a).

**Review by dav3r — APPROVED** [review]
👍 👍

**Pull request merged**